byway/src/main/webapp/recommendations.js

```diff
@@ -0,0 +1,254 @@
+        // by setting a timeout in delayPromise
+        await delayPromise(250);
+        const placesFound = await findPlacesWithTextSearch(request);
+        if(placesFound !== undefined) {
+          addRecommendations(request, placesFound);
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Use the textSearch function from PlacesService to find the results
+ * fitting the request object passed in. Convert callback function into
+ * a chain of promises to return the result directly. If the service
+ * returns a status other than OK, reject the promise and alert the user.
+ * @param {TextSearchRequest} request object with location, radius and query fields.
+ * @return promise either with PlaceResult[] results or undefined if rejected.
+ */
+function findPlacesWithTextSearch(request) {
+  return new Promise((resolve, reject) => {
+    placesService.textSearch(request, (result, status) => {
+      if(status === "OK") {
+        resolve(result);
+      } else {
+        reject(new Error(status));
+      }
+    });
+  }).catch((error) => {
+    alertUser(error.message);
+  });
+}
+
+/**
+ * Provides a message to the user if there was an issue
+ * with their request.
+ * TODO: Make a class to indicate as a "warning" and make text red.
+ * Also, organize on front end.
+ * @param {String} msgFromService status String from placesService
+ */
+function alertUser(msgFromService) {
+  let messageContainer = document.getElementById("message-container");
+  let message = document.createElement('p');
+  if(msgFromService === "OVER_QUERY_LIMIT") {
+    message.innerText += "Showing limited results.\n"
+  } else if(msgFromService === "ZERO_RESULTS") {
+    message.innerText = "Some interests are not in the area.\n"
+  }
+  messageContainer.appendChild(message);
+}
+
+/**
+ * Places markers on the locations found from textSearch.
+ * Temporarily limit the amount of suggestions. Keep track of
+ * places found from interests around a location with savePlacesFromInterests function.
+ * TODO: Store more results and limit on UI with option to "show more"
+ * @param {TextSearchRequest} request with unique location and interest
+ * @param {PlaceResults[]} results places found with PlaceResult type.
+ */
+function addRecommendations(request, placesFound) {
+  const MAX_RECOMMENDATIONS = 1;
+  let placesLoaded = [];
+  for (place of placesFound) {
+    placeMarker(place);
+    placesLoaded.push(place);
+    if(placesLoaded.length == MAX_RECOMMENDATIONS) {
+      break;
+    }
+  }
+  savePlacesFromInterests(request, placesLoaded)
+}
+
+/**
+ * Saves the places generated from interest and location in sessionStorage as key/value pairs.
+ * @param {TextSearchRequest} request with unique location and interest
+ * @param {PlaceResults[]} placesLoaded List of places related to interest
+ */
+function savePlacesFromInterests(request, placesLoaded) {
+  sessionStorage.setItem(JSON.stringify(request), JSON.stringify(placesLoaded));
+}
```

Comment:
Note: When I debugged with the console, it seems like part of how the sessionStorage sets items is not entirely reliable. I got a warning that `open_now()` is deprecated as of Nov. 2019 and will be removed Nov. 2020, and to instead use `is_open()`. I tried to see if there was an update to sessionStorage / alternative, but could not find anything on this.

Reply:
The issue seems to be storing the placesLoaded containing `placeResults` in the sessionStorage. Since it is storing the entire place object, it contains a field, `PlaceResult.opening_hours.open_now`, which is deprecated/willl be replaced. Since we aren't storing this in datastore and instead the current browser session, I think this should be fine?

Reply:
Note that using `stringify` on objects that are part of a class may have odd/unexpected results, especially when converting back from a string (it will end up as a plain JS object, not an instance of the original class). When storing things as JSON you may need/want to convert them to plain objects ("records") containing only the fields you care about.

Reply:
The objects being stringified are requests (key) and PlaceResults[] (value). So far, the properties we have been using are accessed via fields (`.geometry.location`, `.name`, etc) and the plain JS object seems to accomplish this. We can use a LatLng literal object in place of the LatLng custom object from the API, but we can't access functions the same way with the plain JS object. I don't think we are using methods from the classes, but I will keep this in mind/relay with my teammates to be wary of this.

Or rather, do either of you recommend storing this information on the client side a different way? This way works and we aren't using class-specific methods that are lost in JSON translation, but if there's another recommendation I could look more into it.